R/minimize.R

```diff
@@ -101,12 +101,31 @@ minimize <- function(objective, subject_to, initial_design,
             cont_design <- update(cont_design, res2$solution)
             cont_design <- make_tunable(cont_design, n1, n2_pivots)
 
-            return(cont_design)
+            out <- list(
```

Comment:
are res and res2 always well defined? What if we call without post processing? Should be NULL then. Why not return `list(design, design_post_processed, nloptr_return, nloptr_return_post_processed)` ? Might be a bit clumsy, so why not `list(design, details = list(design, design_post, ...))` where design is either the smooth or post-processed design?

Reply:
@MatheMax which solution would u prefer?